app/views/recipes/show.html.erb

```diff
@@ -0,0 +1,163 @@
+          <div class="toggle-indicator"></div>
+        <% end %>
+      </div>
+    <% end %>
+  </div>
+<% end %>
+      
+        
+      </div>
+    </div>
+    <p>Cooking time: <b><%= @recipe.cooking_time_hours %> hours</b></p>
+    <hr style="margin-top: 10px; margin-bottom: 10px;">
+    <p class="fs-5"><%= @recipe.description %></p>
+  </div>
+
+
+  <div id="shopping-list-generate-button">
+    <button type="button"
+    id="generate-shopping-list"
+      class="btn btn-primary"
+      data-bs-toggle="modal"
+      data-bs-target="#myModal"
+      onclick="showModal()">
+        Generate shopping list
+    </button>
+
+    
+
+    <!-- Modal (initially hidden) -->
+    <div class="custom-modal" id="myModal" style="display: none;">
+      <div class="custom-modal-content card">
+        <div class="custom-modal-header">
+          <h5 class="custom-modal-title">Generating Shopping List</h5>
+          <button type="button" class="custom-btn-close" onclick="hideModal()" aria-label="Close"></button>
+        </div>
+        <div class="custom-modal-body">
+          <!-- Add content for your modal here -->
+          <div>
+  <div>
+    <div>
+      
+      <div id="">
+        <h6>Choose an inventory to use</h6>
+        <%= form_with url: shopping_lists_path(recipe_id: @recipe.id), method: :get do |form| %>
+          <div id="modal-input">
+          <div>
+          <%= select("inventory", "inventory_id", Inventory.all.collect { |p| [ p.name, p.id ] }, { prompt: 'Select Inventory' }) %>
+          <%= form.number_field :recipe_id, class: "form-control", value: @recipe.id, hidden: true %>
+          </div>
+      <div>
+        <%= form.submit "Generate", data: { turbo: false }, class: "btn btn-primary generate-submit" %>
+        <% end %>
+      </div>
+      </div>
+      </div>
+    </div>
+  </div>
+</div>
+        </div>
+        <div class="custom-modal-footer">
+          <button type="button" class="btn btn-secondary" onclick="hideModal()">Close</button>
+          <!-- Add additional buttons or actions here -->
+        </div>
+      </div>
+    </div>
+  </div>
+
+
+    
+
+    <% if @recipe.user == current_user %>
+    <button class="add-ingredient-button">
+    <%= link_to "Add ingredient", new_recipe_food_path(recipe_id: @recipe.id) %>
+    </button>
+    <% end %>
+  </div>
+
+  
+    <hr>
+
+
+    
+      <% if @recipe_foods.empty? %>
+        <h4>No foods have been added yet.</h4>
+      <% else %>
+
+      <table>
+    <thead>
+      <tr>
+        <th>Food</th>
+        <th>Quantity</th>
+        <th>Value</th>
+        <th>Actions</th>
+      </tr>
+    </thead>
+    <tbody>
+        <% @recipe_foods.each do | recipe_food | %>
+          <tr>
+        <td><div><%= recipe_food.food.name %></div></td>
+        <td><div><%= recipe_food.quantity %> <%= recipe_food.food.measurement_unit %></div></td>
+        <td><div><b>$<%= recipe_food.food.price_dollars * recipe_food.quantity %></b></div></td>
```

Comment:
- [ ] Same as explained in the previous comment :point_up: This code will also create **N+1** queries problem in your application. Kindly make sure you fix **N+1** queries problem from happening